packages/back-end/src/models/SavedGroupModel.ts

```diff
@@ -0,0 +1,79 @@
+import mongoose from "mongoose";
+import { SavedGroupInterface } from "../../types/saved-group";
+import uniqid from "uniqid";
+
+const savedGroupSchema = new mongoose.Schema({
+  id: {
+    type: String,
+    unique: true,
+  },
+  orgId: {
+    type: String,
+    index: true,
+  },
+  groupName: String,
+  owner: String,
+  dateCreated: Date,
+  dateUpdated: Date,
+  values: [String],
+  attributeKey: String,
+});
+
+type SavedGroupDocument = mongoose.Document & SavedGroupInterface;
+
+const SavedGroupModel = mongoose.model<SavedGroupDocument>(
+  "savedGroup",
+  savedGroupSchema
+);
+
+function parseSaveGroupString(list: string) {
+  const listArr = list.split(",");
+
+  const savedGroup = listArr.map((i: string) => {
+    return i.trim();
+  });
+
+  return [
+    ...new Set(savedGroup.filter((value) => value !== "," && value !== "")),
+  ];
+}
+
+type SavedGroupEditableProps = Omit<
+  SavedGroupInterface,
+  "dateCreated" | "dateUpdated" | "id" | "values"
+>;
+
+export async function createSavedGroup(
+  values: string,
+  group: SavedGroupEditableProps
+): Promise<SavedGroupInterface> {
+  const newGroup = await SavedGroupModel.create({
+    ...group,
+    values: parseSaveGroupString(values),
+    id: uniqid("grp_"),
+    dateCreated: new Date(),
+    dateUpdated: new Date(),
+  });
+  return newGroup.toJSON();
+}
+
+export async function getAllSavedGroups(
+  orgId: string
+): Promise<SavedGroupInterface[]> {
+  const savedGroups = await SavedGroupModel.find({ orgId });
+  return savedGroups.map((value) => value.toJSON()) || [];
+}
+
+export async function updateSavedGroup(
+  values: string,
+  group: SavedGroupEditableProps
+): Promise<void> {
```

Comment:
A few comments about this function:
1. We usually do updates based on an id and not a name field since names can change, but ids can't
2. Your `updateOne` query should query on both organization and id.  Otherwise, if someone guessed another org's saved group id, they might be able to craft an API request to edit it.
3. Instead of have `groupList` as a standalone argument, it makes more sense for the organization and group id to be used instead
    ```ts
    updateSavedGroup(orgId: string, savedGroupId: string, data: SavedGroupEditableProps)
    ```
    This way, the code reads like: "Update group X and change the following properties"
4. I don't think we want to allow editing the `attributeKey`.  If someone is already using the saved group in a feature rule, changing the attributeKey could break things.

Reply:
This feels so much better. Thank you